src/topology/config/mod.rs

```diff
@@ -118,6 +121,32 @@ pub enum DataType {
     Metric,
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+pub struct LogSchemaOptions {
+    #[serde(default = "LogSchemaOptions::default_host_key")]
+    pub host_key: Atom,
+    #[serde(default = "LogSchemaOptions::default_message_key")]
+    pub message_key: Atom,
+    #[serde(default = "LogSchemaOptions::default_timestamp_key")]
+    timestamp_key: Atom,
+}
+
+impl LogSchemaOptions {
+    fn default_host_key() -> Atom { event::HOST.clone() }
+    fn default_message_key() -> Atom { event::MESSAGE.clone() }
+    fn default_timestamp_key() -> Atom { event::TIMESTAMP.clone() }
+}
+
+impl Default for LogSchemaOptions {
+    fn default() -> Self {
+        LogSchemaOptions {
+            host_key: event::HOST.clone(),
+            message_key: event::MESSAGE.clone(),
+            timestamp_key: event::TIMESTAMP.clone(),
+        }
+    }
+}
```

Comment:
We have used the [`Derivative` custom derive macro](https://mcarton.github.io/rust-derivative/Default.html) in other places to reduce this boilerplate. It might be useful here as well.